src/EditorFeatures/Core/Implementation/ExtractMethod/AbstractExtractMethodCommandHandler.cs

```diff
@@ -72,27 +72,28 @@ public VSCommanding.CommandState GetCommandState(ExtractMethodCommandArgs args)
 
         public bool ExecuteCommand(ExtractMethodCommandArgs args, CommandExecutionContext context)
         {
-            var document = args.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
-            if (document == null)
+            using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Applying_Extract_Method_refactoring))
             {
-                return false;
-            }
+                var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoaddedOpenDocumentInCurrentContextWithChangesAsync(
```

Comment:
1. "Loaded". 
2. I can understand the desire to use "InCurrentContextWithChanges" to mirror the existing document call. However, i think it's somewhat redundant unecessary.  It's always been clear that you are getting the Document matching a particular snapshot (so the "with changes" is unnecessary). Also, the concept of 'contexts' isn't something that generally needs to be stated.  So i'd prefer simply GetFullyLoadedDocument.  But i'm not going to heavily argue this.

Reply:
I was also wondering what "current context" and "with changes" means.

---
In reply to: [264542810](https://github.com/dotnet/roslyn/pull/33779#discussion_r264542810) [](ancestors = 264542810)

Reply:
@tmat Unfortunately, late in Roslyn, we added support for SharedProjects.  There was a need to basically say "I have an ITextSnapshot/Buffer, get me the Document".  In a SharedProject world this is ambiguous as a single Snapshot/Buffer could be in *multiple* projects depending on how you expanded this:

![image](https://user-images.githubusercontent.com/4564579/54221660-358f6180-44b1-11e9-9623-5f232bd72396.png)

So, given a Snapshot/Buffer you can ask for *all* the DocumentIds it corresponds to.  But, to make life easy, an entrypoint was given to say "Give me the Document for this that corresponds to the current shared project the user is viewing it in".  And thus was born GetDocumentInCurrentContext.

The 'WithChanges' bit indicates "and the document should correspond exactly to the contents of the snapshot you asked about".  This is important in case the snapshot just changed, and you don't want to get something stale.

--

**UNFORTUNATELY**: it turns out 99% of all use cases are: I want do get teh corresponding Document, and of course its contents should match this snapshot.  So, we really should have just named this `textSnapshot.GetDocument`.  It made me sad we shipped this way, but you know how it goes.  :(